typescript/src/events/guildMessageLog.ts

```diff
@@ -1,29 +1,18 @@
-import { GuildSettings } from '#lib/database';
-import { MessageLogsEnum } from '#utils/constants';
+import type { GuildEntity } from '#lib/database';
 import { Event } from '@sapphire/framework';
+import { isNullish, Nullish, PickByValue } from '@sapphire/utilities';
 import { DiscordAPIError, Guild, HTTPError, MessageEmbed, TextChannel } from 'discord.js';
 
-const TYPES = {
-	[MessageLogsEnum.Member]: GuildSettings.Channels.Logs.Member,
-	[MessageLogsEnum.Message]: GuildSettings.Channels.Logs.Message,
-	[MessageLogsEnum.Image]: GuildSettings.Channels.Logs.Image,
-	[MessageLogsEnum.Moderation]: GuildSettings.Channels.Logs.Moderation,
-	[MessageLogsEnum.NSFWMessage]: GuildSettings.Channels.Logs.NSFWMessage,
-	[MessageLogsEnum.Reaction]: GuildSettings.Channels.Logs.Reaction
-} as const;
-
 export class UserEvent extends Event {
-	public async run(type: MessageLogsEnum, guild: Guild, makeMessage: () => Promise<MessageEmbed> | MessageEmbed) {
-		const key = TYPES[type];
-		if (!key) {
-			this.context.client.logger.warn(`[EVENT] GuildMessageLog: Unknown type '${type}'`);
-			return;
-		}
-
-		const id = await guild.readSettings(key);
-		if (!id) return;
+	public async run(
+		guild: Guild,
+		channelId: string | Nullish,
```

Comment:
```suggestion
		logChannelId: string | Nullish,
```

You should've seen this one coming after all the similar comments